### PR TITLE
Remove named for pfdns

### PR DESCRIPTION
**Pull request by fdurand**
Please review the code

### CODE REVIEW

File: NEWS.old [hunk 1]
```diff
@@ -13,6 +13,20 @@ For more details and developer visible changes see the ChangeLog file.
 For a list of compatibility related changes see the UPGRADE file.
 
 --------------------------------------------------------------------------------
+Version <develVersion> released on <develVersionReleaseDate>
```
Comment (dwlfrth) [inline NEWS.old]: That should go into the new asciidoc based file NEWS.asciidoc
NEWS.old is only for archive purposes

File: conf/pf.conf.defaults [hunk 2]
```diff
@@ -471,7 +471,7 @@ dhcpd_binary=/usr/sbin/dhcpd
 # services.named_binary
 #
 # Location of the named binary. Only necessary to change if you are not running the RPMed version.
-named_binary=/usr/sbin/named
+named_binary=/usr/sbin/pfdns
```
Comment (dwlfrth) [inline conf/pf.conf.defaults]: Wrong

**Comment by dwlfrth** [comment]
Remove news.old from commit :)
Looks like the merged is rewrite all the stuff :)

File: lib/pf/web/util.pm [hunk 3]
```diff
@@ -28,7 +28,10 @@ BEGIN {
     our ( @ISA, @EXPORT );
     @ISA = qw(Exporter);
     # No export to force users to use full package name and allowing pf::web::custom to redefine us
```
Comment (dwlfrth) [inline lib/pf/web/util.pm]: Either remove the exported methods or the comment ;)

File: conf/documentation.conf [hunk 4]
```diff
@@ -237,16 +237,11 @@ EOT
...
 description=<<EOT
 Method by which content is delivered to trapped systems. When set to 
-"proxy", PacketFence uses Apache's reverse proxy functionality and 
-rewrites links. Note that any link to an external resource on the 
-target server will not be properly rewritten. When set to "iptables",
-PacketFence creates passthroughs to the content for only those nodes
-trapped with the corresponding violation.  Be aware that an iptables
-passthrough is based on IP address and clients will be able to get to
-ALL content on the destination site.
+"enabled", PacketFence uses pfdns to resolv the real ip address and
```
Comment (dwlfrth) [inline conf/documentation.conf]: s/resolv/resolve

File: conf/pf.conf.defaults [hunk 5]
```diff
@@ -102,12 +102,8 @@ redirtimer=20s
 #
 # trapping.passthrough
 #
-# Method by which content is delivered to trapped systems. When set to "proxy", PacketFence uses Apache's reverse proxy 
-# functionality and rewrites links. Note that any link to an external resource on the target server will not be properly
-# rewritten.
-# When set to "iptables", PacketFence creates passthroughs to the content for only those nodes trapped with the 
-# corresponding violation.  Be aware that an iptables passthrough is based on IP address and clients will be able to 
-# get to ALL content on the destination site.
+# Method by which content is delivered to trapped systems. When set to "enabled", pfdns will resolv the real ip address
```
Comment (dwlfrth) [inline conf/pf.conf.defaults]: s/resolv/resolve

File: docs/PacketFence_Administration_Guide.asciidoc [hunk 6]
```diff
@@ -2051,7 +2051,7 @@ OAuth2 Authentication
 
 The captive portal of PacketFence allows a guest/user to register using his Google, Facebook or Github account.
 
-For each providers, we maintain an allowed IP list to punch holes into the firewall so the user can hit the provider login page.  This file is located in `conf/oauth2-ips`.  You might need to maintain it quite often, as well as the different DNS zones also located in `conf`.
+For each providers, we maintain an allowed domain list to punch holes into the firewall so the user can hit the provider login page.
```
Comment (dwlfrth) [inline docs/PacketFence_Administration_Guide.asciidoc:2054]: Maybe we can explain how to maintain that list (in the OAuth2 authentication source from the GUI)

File: html/pfappserver/lib/pfappserver/Form/Authentication/Source/Facebook.pm [hunk 7]
```diff
@@ -80,6 +80,19 @@ has_field 'redirect_url' =>
...
+has_field 'domains' =>
+  (
+   type => 'Text',
+   label => 'Authorized domains',
+   required => 1,
+   default => pf::Authentication::Source::FacebookSource->meta->get_attribute('domains')->default,
+   element_attr => {'placeholder' => pf::Authentication::Source::FacebookSource->meta->get_attribute('domains')->default},
+   element_class => ['input-xlarge'],
+   tags => { after_element => \&help,
+             help => 'List of the domain that will be resolv with the correct ip.' },
```
Comment (dwlfrth) [inline html/pfappserver/lib/pfappserver/Form/Authentication/Source/Facebook.pm]: List of domains that will be resolved to IP addresses and will be allowed through the PacketFence firewall to be reached from within the registration VLAN.

File: html/pfappserver/lib/pfappserver/Form/Authentication/Source/Github.pm [hunk 8]
```diff
@@ -77,6 +77,17 @@ has_field 'redirect_url' =>
...
+has_field 'domains' =>
+  (
+   type => 'Text',
+   label => 'Authorized domains',
+   required => 1,
+   default => pf::Authentication::Source::GithubSource->meta->get_attribute('domains')->default,
+   element_attr => {'placeholder' => pf::Authentication::Source::GithubSource->meta->get_attribute('domains')->default},
+   element_class => ['input-xlarge'],
+   tags => { after_element => \&help,
+             help => 'List of the domain that will be resolv with the correct ip.' },
```
Comment (dwlfrth) [inline html/pfappserver/lib/pfappserver/Form/Authentication/Source/Github.pm]: List of domains that will be resolved (using pfdns) to IP addresses and will be allowed through the PacketFence firewall to be reached from within the registration VLAN.

File: html/pfappserver/lib/pfappserver/Form/Authentication/Source/Google.pm [hunk 9]
```diff
@@ -90,6 +90,17 @@ has_field 'redirect_url' =>
...
+has_field 'domains' =>
+  (
+   type => 'Text',
+   label => 'Authorized domains',
+   required => 1,
+   default => pf::Authentication::Source::GoogleSource->meta->get_attribute('domains')->default,
+   element_attr => {'placeholder' => pf::Authentication::Source::GoogleSource->meta->get_attribute('domains')->default},
+   element_class => ['input-xlarge'],
+   tags => { after_element => \&help,
+             help => 'List of the domain that will be resolv with the correct ip.' },
```
Comment (dwlfrth) [inline html/pfappserver/lib/pfappserver/Form/Authentication/Source/Google.pm]: List of domains that will be resolved (using pfdns) to IP addresses and will be allowed through the PacketFence firewall to be reached from within the registration VLAN.

File: lib/pf/ipset.pm [hunk 10]
```diff
@@ -69,6 +69,21 @@ sub iptables_generate {
...
+    my $github_enabled = $guest_self_registration{$SELFREG_MODE_GITHUB};
+    my $passthrough_enabled = isenabled($Config{'trapping'}{'passthrough'});
+
+    if ($google_enabled || $facebook_enabled || $github_enabled || $passthrought_enabled) {
+        if ($IPSET_VERSION > 4) {
+            $cmd = "LANG=C sudo ipset --create pfsession_oauth hash:ip,port 2>&1";
+             my @lines  = pf_run($cmd);
+        }
+        else {
+            $logger->warn("We doesnt support ipset under version 4");
```
Comment (dwlfrth) [inline lib/pf/ipset.pm]: We do not support ipset version lower than version 4

File: docs/PacketFence_Administration_Guide.asciidoc (hunk 6 above)
Comment (fdurand) [inline docs/PacketFence_Administration_Guide.asciidoc:2054]: Ok i have added a line under to explain

**Pull request merged**